### ISSUE TITLE
IPTC metadata are not encoded in UTF-8

### ISSUE THREAD

**Issue by fredjoncour** [darktable-org/darktable#16797]
### Describe the bug

IPTC metadata exports don't seem to support UTF-8 (french accents. Ex french accent "é" is replaced by "@&..." symbols). Unfortunately, I can't export metadata in XMP, as this format is not supported by the platform through which I distribute and sell my work. Thank you very much.
Frédéric

### Steps to reproduce

No specific action. I export my IPTC fields using the IPTC.Application2 variables in the export module.

### Where did you obtain darktable from?

downloaded from www.darktable.org

### darktable version

4.6.1

### What OS are you using?

Windows

### What is the version of your OS?

10

### Describe your system?

When I transfer my images to my FTP server (set to UTF-8), all accents are replaced by symbols. This problem does not occur with Lightroom 6.14 (on the same PC).

### Are you using OpenCL GPU in darktable?

No

**Comment by zisoft** [comment]
Can you please give some more information here, because I cannot reproduce that.
Here is what I tried:

Set the metadata title of a photo:
![Bildschirmfoto 2024-05-15 um 18 13 20](https://github.com/darktable-org/darktable/assets/4083281/8309ca79-9346-4cf3-b0be-abe10eba2fdb)

Export settings:
![Bildschirmfoto 2024-05-15 um 18 13 36](https://github.com/darktable-org/darktable/assets/4083281/d167e896-eeb5-4f5b-9996-ffd32c6b7250)

Result in the exported image:
```
exiv2 -p i 20231110_204315_Dresden_0052.jpg
Iptc.Application2.Caption                    String     19  Test for Frédéric
```

**Comment by zisoft** [comment]
Can reproduce on Windows

**Comment by fredjoncour** [comment]
Hello Zisoft,
My photos are distributed on the PixPalace platform, which brings together the biggest French photo agencies such as Magnum and SIPAPress. The only protocol accepted on this platform for transferring metadata into their database is IPTC. Since recently, I've been using Darktable. I used to use Lightroom 6.14.
After entering all my metadata into Darktable, I export my images and transfer them to the platform.
![metadataDT-F Joncour](https://github.com/darktable-org/darktable/assets/166613702/31da2d09-5b25-4b38-9bc8-4875b78aa164)
![export-DT-F Joncour](https://github.com/darktable-org/darktable/assets/166613702/7f59a76d-acff-4d46-919f-1a71f9989563)

When I read my metadata in The Gimp or with Windows properties ("details" tab), accents look good!
Then I upload my images to PixPalace using Filezilla (in UTF-8 mode). This is where my accents are replaced by symbols.
When I did the same thing with Lightroom, using the same image and copying/pasting the metadata from DT to Lr, there was no problem with accents once the photo was uploaded to the PixPalace FTP server.
Thank you for your attention.

**Comment by zisoft** [comment]
Please try again with `Iptc.Envelope.CharacterSet` set to the Escape sequence `ESC % G` which stands for UTF8.

No idea how to enter this via keyboard, so please open the attached text file and copy/paste the sequence.

[UTF8_ESC_Sequence.txt](https://github.com/darktable-org/darktable/files/15334387/UTF8_ESC_Sequence.txt)

Should look like this:

![Screenshot 2024-05-16 130736](https://github.com/darktable-org/darktable/assets/4083281/90269890-a5db-443b-9834-06dc4c139cb4)

Then try again.

**Comment by zisoft** [comment]
@fredjoncour: I have created a PR which will set this by default on export (if not otherwise specified by the user).

**Comment by fredjoncour** [comment]
Thank you very much,
Accents are ok, thank you. However, this new setting cancels the export of the value iptc.application2.transmissionReference | $(CATEGORY0(subjects)). I get an empty field. See image :
![export-DT-F Joncour-with-correct](https://github.com/darktable-org/darktable/assets/166613702/205ee6b2-7f70-4cb0-a389-281decfef14f)

**Comment by zisoft** [comment]
Cannot reproduce here, transmissionReference gets exported as expected.
And I cannot see why there should be an influence by Envelope.CharacterSet.

**Comment by fredjoncour** [comment]
Ok, I try again. Thank

**Comment by fredjoncour** [comment]
I'm a donkey!!! I had changed the value of "$(CATEGORY0(sujets))" in the keywords without reporting the changes in the export module !!! It's ok, fantastic !!! I'm trying to upload a new file to the FTP server and I'll keep you posted! Thanks a lot.
Frédéric

**Comment by zisoft** [comment]
If @TurboGit accepts this as a bugfix it will be corrected in the upcoming version 4.8 to be released in June.
Otherwise it will be part of version 5.0 in December. In this case you can use the workaround above.
